soh/soh/Enhancements/randomizer/draw.cpp

```diff
@@ -0,0 +1,116 @@
+        { 80, 80, 80 }     // Ganon's Castle
+    };
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_80093D18(globalCtx->state.gfxCtx);
+
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, (char*)__FILE__, __LINE__),
+              G_MTX_MODELVIEW | G_MTX_LOAD);
+
+    gsDPSetGrayscaleColor(POLY_OPA_DISP++, colors[color_slot][0], colors[color_slot][1], colors[color_slot][2], 255);
+    gsSPGrayscale(POLY_OPA_DISP++, true);
+
+    gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gGiSmallKeyDL);
+
+    gsSPGrayscale(POLY_OPA_DISP++, false);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+extern "C" void Randomizer_DrawBossKey(GlobalContext* globalCtx, GetItemEntry* getItemEntry) {
+    s32 pad;
+    s16 color_slot;
+    color_slot = getItemEntry->getItemId - RG_FOREST_TEMPLE_BOSS_KEY;
+    s16 colors[6][3] = {
+        { 4, 195, 46 },   // Forest Temple
+        { 237, 95, 95 },  // Fire Temple
+        { 85, 180, 223 }, // Water Temple
+        { 222, 158, 47 }, // Spirit Temple
+        { 126, 16, 177 }, // Shadow Temple
+        { 210, 0, 0 }     // Ganon's Castle
+    };
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_80093D18(globalCtx->state.gfxCtx);
+
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, (char*)__FILE__, __LINE__),
+              G_MTX_MODELVIEW | G_MTX_LOAD);
+
+    if (color_slot == 5) { // Ganon's Boss Key
+        gsDPSetGrayscaleColor(POLY_OPA_DISP++, 80, 80, 80, 255);
+        gsSPGrayscale(POLY_OPA_DISP++, true);
+    }
+
+    gSPDisplayList(POLY_OPA_DISP++, (Gfx*)gGiBossKeyDL);
+
+    if (color_slot == 5) { // Ganon's Boss Key
+        gsSPGrayscale(POLY_OPA_DISP++, false);
+    }
+
+    func_80093D84(globalCtx->state.gfxCtx);
+
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, (char*)__FILE__, __LINE__),
+              G_MTX_MODELVIEW | G_MTX_LOAD);
+
+    if (color_slot >= 0) {
+        gsDPSetGrayscaleColor(POLY_XLU_DISP++, colors[color_slot][0], colors[color_slot][1], colors[color_slot][2],
+                              255);
+        gsSPGrayscale(POLY_XLU_DISP++, true);
+    }
+
+    gSPDisplayList(POLY_XLU_DISP++, (Gfx*)gGiBossKeyGemDL);
+
+    if (color_slot >= 0) {
+        gsSPGrayscale(POLY_XLU_DISP++, false);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+extern "C" void Randomizer_DrawDoubleDefense(GlobalContext* globalCtx, GetItemEntry getItemEntry) {
+    s32 pad;
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_80093D84(globalCtx->state.gfxCtx);
+   
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, (char*)__FILE__, __LINE__), G_MTX_MODELVIEW | G_MTX_LOAD);
+
+    // if (drawId == doubleDef) {
+    gsDPSetGrayscaleColor(POLY_XLU_DISP++, 255, 255, 255, 255);
+    gsSPGrayscale(POLY_XLU_DISP++, true);
+    // }
+
+    gSPDisplayList(POLY_XLU_DISP++, (Gfx*)gGiHeartBorderDL);
+
+    // if (drawId == doubleDef) {
+    gsSPGrayscale(POLY_XLU_DISP++, false);
+    // }
```

Comment:
Can remove the comments since these also aren't needed considering Double Defense has its own function now.